pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -633,3 +706,203 @@ func jivaUpgrade(pvName, openebsNamespace string, utaskObj *utask.UpgradeTask) (
 	klog.Info("Upgrade Successful for", pvName)
 	return options.utaskObj, nil
 }
+
+func (j *jivaVolumeOptions) cleanup(openebsNamespace string) error {
+	var err error
+	if j.replicaObj.version == currentVersion && j.replicaObj.name != "" {
+		klog.Info("cleaning old replica deployment")
+		err = deployClient.WithNamespace(j.ns).Delete(j.replicaObj.name, &metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	if j.controllerObj.version == currentVersion && j.ns != openebsNamespace {
+		klog.Info("cleaning old replica deployment")
+		err = deployClient.WithNamespace(j.ns).Delete(j.controllerObj.name, &metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (j *jivaVolumeOptions) migrate(pvName, openebsNamespace string) error {
+	err := j.migrateReplica(openebsNamespace)
+	if err != nil {
+		return err
+	}
+	err = j.migrateTarget(pvName, openebsNamespace)
+	return err
+}
+
+func (j *jivaVolumeOptions) migrateReplica(openebsNamespace string) error {
+	// get the separate replica deployments in openebs namespace
+	deployList, err := deployClient.WithNamespace(openebsNamespace).List(&metav1.ListOptions{
+		LabelSelector: replicaDeployLabel,
+	})
+	if err != nil {
+		return err
+	}
+	// get the old replica deployment by name
+	oldDeployObj, err := deployClient.WithNamespace(j.ns).Get(j.replicaObj.name)
+	if err != nil {
+		return err
+	}
+	replicaCount := int(*oldDeployObj.Spec.Replicas)
+	replicasLeft := len(deployList.Items)
+	// if the volume was deployed in openebs namespace while provisioning
+	// as the old deployment also has the same label
+	if j.ns == openebsNamespace {
+		replicasLeft = replicasLeft - 1
+	}
+	if len(deployList.Items) != replicaCount {
+		klog.Infof("splitting replica deployment")
+		nodeNames := oldDeployObj.Spec.Template.Spec.Affinity.NodeAffinity.
+			RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Values
+		var zero int32
+		for i := replicasLeft; i < replicaCount; i++ {
+			replicaDeploy := oldDeployObj.DeepCopy()
+			replicaDeploy.Name = replicaDeploy.Name + "-" + strconv.Itoa(i+1)
+			replicaDeploy.Namespace = openebsNamespace
+			replicaDeploy.ResourceVersion = ""
+			replicaDeploy.Spec.Replicas = &zero
+			replicaDeploy.Spec.Template.Spec.Affinity.NodeAffinity.
+				RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Values = []string{nodeNames[i]}
+			klog.Infof("creating replica deployment %s in %s namespace", replicaDeploy.Name, openebsNamespace)
+			replicaDeploy, err := deployClient.WithNamespace(openebsNamespace).Create(replicaDeploy)
+			if err != nil {
```

Comment:
by handling 'IsAlreadyExists' error, we can just keep creating on every iteration without much logic in getting the count.
Current code of getting count of replicas to be created is also fine.